lib/dataClasses/seed_response.dart

```diff
@@ -0,0 +1,43 @@
+import 'dart:convert';
+
+/// The SeedResponse class represents the data from a response from the seed
+/// server. It accepts input for deserialization. The factory constructors will
+/// not pass empty or default versions of the class. Rather, they throw
+/// expcetions if the data passed in are invalid.
+class SeedResponse {
+  final String seed;
+  final DateTime expiration;
+
+  SeedResponse({
+    required this.seed,
+    required this.expiration,
+  });
+
+  /// fromJsonString parses all values from the JSON data object passed in and
+  /// attempts to create a SeedResponse object. A couplt things to note:
+  /// This constructor throws exceptions if the data is invalid.
+  /// I didn't use a generator because the Flutter page itself states:
+  /// 'Use manual serialization for smaller projects'
+  /// https://flutter.dev/docs/development/data-and-backend/json
+  factory SeedResponse.fromJsonString(String jsonStr) {
```

Comment:
**kudos:** thank you for stating why you used manual serialization here.

**question:** have you used anything that automatically handles that for you in Dart?

Reply:
I have used a couple JSON parsing generators in the past. I had encountered a few reasons in the past that caused me to handle most JSON parsing manually:

- Sometimes the results I get back from APIs are inconsistent. For instance, a value may be a number or a string representation of a number, true or 'true' and sometimes collections of values (List of numbers). This frequently threw a wrench into the works. 
- Early development frequently required that I continuously regenerate the JSON parsers, while figuring out the best way to handle data responses. Eventually I stopped trying to regen as manually updating the data classes was a bit easier.
- I frequently persist data from these APIs and I would frequently have toJson and fromJson factories that I could feed into each other and have consistent results.